collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2643,11 +2646,20 @@ public void onNumberPickerValueSelected(int widgetId, int value) {
     }
 
     @Override
-    public void onDateChanged(LocalDateTime date) {
-        ODKView odkView = getCurrentViewIfODKView();
-        if (odkView != null) {
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        if (getCurrentViewIfODKView() != null) {
             QuestionWidget widgetGettingNewValue = getWidgetWaitingForBinaryData();
-            setBinaryWidgetData(date);
+            setBinaryWidgetData(DateTimeUtils.getLocalDateTime(
```

Comment:
This method name also should be changed like you did with setData() in this case it might be setWidgetData() without `binary` word because as i said before it is no longer used only by binary widgets.

Reply:
Also the `BinaryDataReceiver` class has wrong name again it's not just for binary widgets.

Reply:
I renamed it to `WidgetDataReceiver`. Is it fine?

Reply:
Yes seems fine.